src/flow2supera/reader.py

```diff
@@ -183,7 +183,7 @@ def GetEventTruthFromHits(self, backtracked_hits, segments, trajectories):
              
                 #check consistency of event id
                 if (trajectory['event_id'] != event_id): 
-                    print("Event IDs of trajectory and segment are different")
+                    print("Event IDs of trajectory and segment are different",trajectory['event_id'],event_id,vertex_id,trajectory['file_traj_id'])
                     raise ValueError
```

Comment:
I would just remove this entire thing since you are already adding the mask. It does not make much sense now

Reply:
I think it is fine to leave in there to catch more errors. We still have the same error later on at the time of this comment.